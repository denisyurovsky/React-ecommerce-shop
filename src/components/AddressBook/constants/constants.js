export const LISTBOX_PADDING = 8;
export const DOUBLE_LISTBOX_PADDING = LISTBOX_PADDING * 2;
export const LIMIT_NUMBER_ITEMS = 8;
export const ITEM_SIZE_LG = 48;
export const ITEM_SIZE_MD = 36;
