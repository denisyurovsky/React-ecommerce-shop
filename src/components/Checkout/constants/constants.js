export const PANEL = {
  DELIVERY_ADDRESS: 'Delivery Address',
  PERSONAL_INFORMATION: 'Personal Information',
  PERSONAL_ADDRESS: 'Personal Address',
  PAYMENT_METHOD: 'Payment method',
  ORDER_CONFIRMATION: 'Order confirmation',
};
