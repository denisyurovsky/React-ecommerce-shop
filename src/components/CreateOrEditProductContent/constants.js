export const PRICE_ERROR = 'Only numbers';
export const DESCRIPTION_ERROR = 'Only latin characters';
export const LENGTH_ERROR = 'Should contain at least one symbol';
export const DISCOUNT_ERROR = 'Discount should be less than price';
export const SELECTION_ERROR = 'Choose category';
