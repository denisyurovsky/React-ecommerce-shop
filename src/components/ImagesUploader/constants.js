export const LOSS_OF_IMAGES = 'We lost images';

export const STATUS = {
  CROPPED: 'cropped',
  CROPPING: 'cropping',
  IDLE: 'idle',
};

export const TEST_IMG_URL = 'http://someURL';
