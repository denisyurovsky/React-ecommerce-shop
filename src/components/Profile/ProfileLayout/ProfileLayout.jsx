import { Card, Container } from '@mui/material';
import Proptypes from 'prop-types';
import React from 'react';

import useBreakPoint from '../../../hooks/useBreakPoint';
import ProfileHeader from '../../Profile/ProfileHeader/ProfileHeader';
import ProfileSidebar from '../../Profile/ProfileSidebar/ProfileSidebar';

import styles from './ProfileLayout.module.scss';

const ProfileLayout = ({ children, title }) => {
  const isLessThenBreakPoint = useBreakPoint();

  return (
    <div className={styles.wrapper}>
      <ProfileHeader title={title} />

      <Container className={styles.container}>
        {!isLessThenBreakPoint ? (
          <>
            <ProfileSidebar />
            <Card className={styles.profile}>{children}</Card>
          </>
        ) : (
          <> {children} </>
        )}
      </Container>
    </div>
  );
};

ProfileLayout.propTypes = {
  children: Proptypes.oneOfType([
    Proptypes.arrayOf(Proptypes.node),
    Proptypes.node,
  ]).isRequired,
  title: Proptypes.string,
};

export default ProfileLayout;
