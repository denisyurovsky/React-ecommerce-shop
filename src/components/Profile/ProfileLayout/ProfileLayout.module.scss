@import "../../../assets/styles/breakpoints.scss";

.wrapper {
  width: 98%;
  padding: 0 0.5rem;
}

.container {
  display: flex !important;
  justify-content: space-between;
  padding: 0 !important;
}

.sidebar {
  position: sticky;
}

.profile {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 75%;
  padding: 4rem 2rem 6rem;
}
