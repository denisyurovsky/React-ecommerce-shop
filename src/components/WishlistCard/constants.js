export const DEFAULT_LABEL_INPUT = 'Wishlist Name';
export const ERROR_LABEL_INPUT = 'Wishlist name should be unique';
