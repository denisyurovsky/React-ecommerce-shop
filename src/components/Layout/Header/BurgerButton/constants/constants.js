import { pathNames } from '../../../../../constants/pathNames';

export const LINKS = [
  {
    name: 'HOME',
    link: '/',
  },
  {
    name: 'PRODUCTS',
    link: pathNames.PRODUCTS,
  },
];
