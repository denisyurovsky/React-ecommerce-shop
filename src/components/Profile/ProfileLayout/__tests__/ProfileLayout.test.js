import React from 'react';

import { BREAK_POINT } from '../../../../constants/constants';
import users from '../../../../test-utils/dto/usersDto';
import render, { screen } from '../../../../test-utils/renderWith';
import ProfileLayout from '../ProfileLayout';

localStorage.setItem('accessToken', 'eyJhbGciOiJIU');

const MyProfileLayout = () => (
  <ProfileLayout profile={users[1].user}>
    <div>Children test</div>
  </ProfileLayout>
);

describe('ProfileLayout component', () => {
  describe('profile sidebar', () => {
    it('should be displayed if width more breakpoint', () => {
      window.innerWidth = BREAK_POINT.LAPTOP + 10;
      const { asFragment } = render(<MyProfileLayout />);

      expect(asFragment()).toMatchSnapshot();
      expect(screen.getByText('My Wishlist')).toBeInTheDocument();
      expect(screen.getByText('Children test')).toBeInTheDocument();
    });

    it('should not be displayed if width equal breakpoint', () => {
      window.innerWidth = BREAK_POINT.LAPTOP;
      render(<MyProfileLayout />);

      expect(screen.queryByText('My profile')).not.toBeInTheDocument();
    });

    it('should not be displayed if width less breakpoint', () => {
      window.innerWidth = BREAK_POINT.LAPTOP - 10;
      render(<MyProfileLayout />);

      expect(screen.queryByText('My profile')).not.toBeInTheDocument();
      expect(screen.getByText('Children test')).toBeInTheDocument();
    });
  });
});
