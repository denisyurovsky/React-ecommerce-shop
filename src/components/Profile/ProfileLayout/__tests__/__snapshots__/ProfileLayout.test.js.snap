// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileLayout component profile sidebar should be displayed if width more breakpoint 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div
      class="breadcrumbs"
      role="presentation"
    >
      <nav
        aria-label="breadcrumb"
        class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
      >
        <ol
          class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
        >
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              class="breadcrumb"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            aria-hidden="true"
            class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
          >
            /
          </li>
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              class="breadcrumb"
              href="/users"
            >
              Users
            </a>
          </li>
          <li
            aria-hidden="true"
            class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
          >
            /
          </li>
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              class="breadcrumb"
              href="/profile"
            >
              Profile
            </a>
          </li>
        </ol>
      </nav>
    </div>
    <div
      class="header"
    >
      <h2
        class="header"
        data-testid="title"
      />
    </div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg container css-1oqqzyl-MuiContainer-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root container css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <ul
          class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
          data-testid="menuList"
          id="composition-menu"
          role="menu"
          tabindex="-1"
        >
          <li
            class="MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
            role="menuitem"
            tabindex="0"
          >
            <a
              class="link grey"
              href="/profile"
              style="opacity: 1;"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="HomeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                />
              </svg>
              <span
                class="menuTitle"
              >
                My Profile
              </span>
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </li>
          <li
            class="MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
            role="menuitem"
            tabindex="-1"
          >
            <a
              class="link grey"
              href="/profile/wishlist"
              style="opacity: 1;"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="FavoriteIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m12 21.35-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
                />
              </svg>
              <span
                class="menuTitle"
              >
                My Wishlist
              </span>
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </li>
          <li
            class="MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
            role="menuitem"
            tabindex="-1"
          >
            <a
              class="link grey"
              href="/profile/orders"
              style="opacity: 1;"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="Inventory2Icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 2H4c-1 0-2 .9-2 2v3.01c0 .72.43 1.34 1 1.69V20c0 1.1 1.1 2 2 2h14c.9 0 2-.9 2-2V8.7c.57-.35 1-.97 1-1.69V4c0-1.1-1-2-2-2zm-5 12H9v-2h6v2zm5-7H4V4h16v3z"
                />
              </svg>
              <span
                class="menuTitle"
              >
                My Orders
              </span>
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </li>
          <li
            class="MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
            role="menuitem"
            tabindex="-1"
          >
            <a
              class="link grey"
              href="/profile/address-book"
              style="opacity: 1;"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ImportContactsIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M17.5 4.5c-1.95 0-4.05.4-5.5 1.5-1.45-1.1-3.55-1.5-5.5-1.5S2.45 4.9 1 6v14.65c0 .65.73.45.75.45C3.1 20.45 5.05 20 6.5 20c1.95 0 4.05.4 5.5 1.5 1.35-.85 3.8-1.5 5.5-1.5 1.65 0 3.35.3 4.75 1.05.41.21.75-.19.75-.45V6c-1.49-1.12-3.63-1.5-5.5-1.5zm3.5 14c-1.1-.35-2.3-.5-3.5-.5-1.7 0-4.15.65-5.5 1.5V8c1.35-.85 3.8-1.5 5.5-1.5 1.2 0 2.4.15 3.5.5v11.5z"
                />
              </svg>
              <span
                class="menuTitle"
              >
                My Address book
              </span>
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </li>
        </ul>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root profile css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <div>
          Children test
        </div>
      </div>
    </div>
  </div>
  <div
    class="Toastify"
  />
</DocumentFragment>
`;
